internal/encryption/pkcs11_test.go

```diff
@@ -0,0 +1,187 @@
+	if _, err := rand.Read(key); err != nil {
+		return nil, fmt.Errorf("failed to generate test key: %v", err)
+	}
+	return &mockPKCS11Context{key: key}, nil
+}
+
+func (m *mockPKCS11Context) Initialize() error                { return nil }
+func (m *mockPKCS11Context) Finalize() error                  { return nil }
+func (m *mockPKCS11Context) GetSlotList(bool) ([]uint, error) { return []uint{0}, nil }
+func (m *mockPKCS11Context) OpenSession(uint, uint) (pkcs11.SessionHandle, error) {
+	return pkcs11.SessionHandle(1), nil
+}
+func (m *mockPKCS11Context) CloseSession(pkcs11.SessionHandle) error        { return nil }
+func (m *mockPKCS11Context) Login(pkcs11.SessionHandle, uint, string) error { return nil }
+func (m *mockPKCS11Context) Logout(pkcs11.SessionHandle) error              { return nil }
+func (m *mockPKCS11Context) FindObjectsInit(pkcs11.SessionHandle, []*pkcs11.Attribute) error {
+	return nil
+}
+func (m *mockPKCS11Context) FindObjects(pkcs11.SessionHandle, int) ([]pkcs11.ObjectHandle, bool, error) {
+	return []pkcs11.ObjectHandle{pkcs11.ObjectHandle(1)}, false, nil
+}
+func (m *mockPKCS11Context) FindObjectsFinal(pkcs11.SessionHandle) error { return nil }
+
+func (m *mockPKCS11Context) EncryptInit(sh pkcs11.SessionHandle, mech []*pkcs11.Mechanism, obj pkcs11.ObjectHandle) error {
+	if len(mech) > 0 && mech[0].Parameter != nil {
+		m.iv = mech[0].Parameter
+	}
+	return nil
+}
+
+func (m *mockPKCS11Context) Encrypt(sh pkcs11.SessionHandle, message []byte) ([]byte, error) {
+	if m.iv == nil {
+		return nil, fmt.Errorf("IV not set")
+	}
+	block, err := aes.NewCipher(m.key)
+	if err != nil {
+		return nil, err
+	}
+
+	padding := aes.BlockSize - len(message)%aes.BlockSize
+	if padding > 0 {
+		message = append(message, make([]byte, padding)...)
+	}
+
+	ciphertext := make([]byte, len(message))
+	mode := cipher.NewCBCEncrypter(block, m.iv)
+	mode.CryptBlocks(ciphertext, message)
+
+	return ciphertext, nil
+}
+
+func (m *mockPKCS11Context) DecryptInit(sh pkcs11.SessionHandle, mech []*pkcs11.Mechanism, obj pkcs11.ObjectHandle) error {
+	if len(mech) > 0 && mech[0].Parameter != nil {
+		m.iv = mech[0].Parameter
+	}
+	return nil
+}
+
+func (m *mockPKCS11Context) Decrypt(sh pkcs11.SessionHandle, message []byte) ([]byte, error) {
+	if m.iv == nil {
+		return nil, fmt.Errorf("IV not set")
+	}
+	block, err := aes.NewCipher(m.key)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(message)%aes.BlockSize != 0 {
+		return nil, fmt.Errorf("ciphertext is not a multiple of the block size")
+	}
+
+	plaintext := make([]byte, len(message))
+	mode := cipher.NewCBCDecrypter(block, m.iv)
+	mode.CryptBlocks(plaintext, message)
+
+	for i := len(plaintext) - 1; i >= 0; i-- {
+		if plaintext[i] != 0 {
+			return plaintext[:i+1], nil
+		}
+	}
+	return plaintext, nil
+}
+
+func TestPKCS11Backend_EncryptDecryptEndToEnd(t *testing.T) {
+	tests := []struct {
+		name      string
+		message   []byte
+		wantErr   bool
+		errorCase string
```

Comment:
This field is unclear to me, why do we capture "short" independently, and have a switch case for it. Does this really belong in a table test if we need switch cases?